boards: arm: nucleo_l152re: add support for spi

HoZHel opened the pull request:
Defined SPI1, SPI2, and SPI3 pin controls for Nucleo-L152RE.
Defined arduino_spi.
Add definition of SPI3 for STM32L1 series.

HoZHel committed:
dts: arm: st: l1: add spi3

Add definition of SPI3 for STM32L1xxxC/D/E series.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

erwango commented on dts/arm/st/l1/stm32l1.dtsi:
```diff
@@ -200,6 +200,16 @@
 			status = "disabled";
 		};
 
+    spi3: spi@40003c00 {
```
Please fix indentation

HoZHel replied:
Done. Thanks, see also https://github.com/zephyrproject-rtos/zephyr/pull/61656#discussion_r1299861203

gautierg-st commented on dts/arm/st/l1/stm32l1.dtsi:
```diff
@@ -200,6 +200,16 @@
 			status = "disabled";
 		};
 
+    spi3: spi@40003c00 {
```
SPI3 is not available on the complete STM32L1x series but only on STM32L1xxxC/D/E.
So this definition should be added in stm32l151Xc.dtsi, stm32l152Xc.dtsi and stm32l152Xe.dtsi (like it is done for timers5 for example)

HoZHel replied:
Done. Thanks for the suggestion.

HoZHel committed:
boards: arm: nucleo_l152re: add support for spi

Defined SPI1, SPI2, and SPI3 pin controls for Nucleo-L152RE.
Defined arduino_spi.

Signed-off-by: Ali Hozhabri <ali.hozhabri@st.com>

gautierg-st reviewed (approved):
Thanks
